File: common/src/main/java/io/muun/common/model/CreateSessionRcOk.java
package io.muun.common.model;

import io.muun.common.Optional;
import io.muun.common.api.KeySet;

import javax.annotation.Nullable;

public class CreateSessionRcOk {

    private final Optional<KeySet> keySet; // Empty if hasEmailSetup = true
    private final boolean hasEmailSetup;
    private final Optional<String> obfuscatedEmail;  // Empty if if hasEmailSetup = false

    /**
     * Constructor.
     */
    public CreateSessionRcOk(@Nullable KeySet keySet,
                             boolean hasEmailSetup,
                             @Nullable String obfuscatedEmail) {

        this.keySet = Optional.ofNullable(keySet);
        this.hasEmailSetup = hasEmailSetup;
        this.obfuscatedEmail = Optional.ofNullable(obfuscatedEmail);
    }

    public Optional<KeySet> getKeySet() {
        return keySet;
    }

    /**
     * Getter for hasEmailSetup.
     */
    public boolean hasEmailSetup() {
        return hasEmailSetup;
    }

    public Optional<String> getObfuscatedEmail() {
        return obfuscatedEmail;
    }
}

File: android/apolloui/src/main/java/io/muun/apollo/domain/model/CreateSessionRcOk.kt
package io.muun.apollo.domain.model

import io.muun.common.api.KeySet

data class CreateSessionRcOk(
    val keySet: KeySet?,        // Null if hasEmailSetup = true
    @JvmField val hasEmailSetup: Boolean,
    val obfuscatedEmail: String?,    // Null if hasEmailSetup = false
    val playIntegrityNonce: String?,
)